druntime/Makefile

```diff
@@ -393,7 +393,7 @@ $(ROOT)/errno_c$(DOTOBJ) : src/core/stdc/errno.c $(DMD)
 	@mkdir -p $(dir $@)
 	$(DMD) -c $(DFLAGS) -I. -P=-I. $< -of$@
 
-$(ROOT)/threadasm$(DOTOBJ) : src/core/thread/fiber/switch_context_asm.S
+$(ROOT)/threadasm$(DOTOBJ) : src/core/thread/fiber/switch_context_asm.S src/core/thread/fiber/switch_context_riscv.S
```

Comment:
Hey Denis, first of all sorry for this PR being ignored for so long. I can't really comment on the asm itself, but I take it you've tested it, so I think that's a start at the very least.

I don't think this change here works. `$<` in the recipe is the first input, so the new asm file would be ignored. You'd most likely either need a 2nd rule, or simply merge the new asm into the existing file, as for the existing archs needing asm. I know you don't like it ;) - but this is code hardly anyone will have to look at, so the file doesn't need to be super-pretty, and keeping it a single file means no build systems need to be adapted.

Reply:
What if just don't change this line at all?

Actually, DMD won't use this file because it doesn't compile for RISC-V, LDC uses its own CMake scripts and GDC uses another approach at all for context switching code organisation (https://github.com/gcc-mirror/gcc/tree/master/libphobos/libdruntime/config)

Reply:
>  $< in the recipe is the first input, so the new asm file would be ignored. 

I checked it right now just by swapping these two .S files and druntime tests passed.
So, seems, outside of the Make target `$(ROOT)/threadasm$(DOTOBJ)` nobody knows that target contains two files

Reply:
No it doesn't work at all, as I said: https://github.com/dlang/dmd/actions/runs/14878562144/job/41780986635?pr=21353#step:10:78:
> cc -c -m64 -fPIC -DHAVE_UNISTD_H -O3 --target=x86_64-darwin-apple   src/core/thread/fiber/switch_context_riscv.S -o../generated/osx/release/64/threadasm.o

That 2nd PR of yours just proves that DMD doesn't need the current existing file at all.

Reply:
Well if you want to add it for both LDC and GDC, then here upstream is the place to add it.

Reply:
GCC will do: they approach assumes splitting this ASM file into pieces for each architecture: https://github.com/gcc-mirror/gcc/tree/master/libphobos/libdruntime/config (Or maybe they wrote context switching from scratch, I don't really know.)

Reply:
Some have likely diverged to support more platforms and assemblers.

Not everything in assembly is for fiber support. SystemZ has an asm implementation of get_tls_offset for example.